test/prototype/mx_formats/test_e3m0.py

```diff
@@ -0,0 +1,188 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest
+
+import torch
+from torchao.prototype.mx_formats.constants import (
+    F32_MIN_NORMAL,
+    F4_E3M0_EXP_BIAS,
+    F4_E3M0_MAX,
+    F4_E3M0_MIN_NORMAL,
+)
+from torchao.prototype.mx_formats.custom_cast import (
+    EBITS_F4_E3M0,
+    f32_to_f4_e3m0_unpacked,
+    MBITS_F4_E3M0,
+)
+from torchao.prototype.mx_formats.mx_tensor import MXTensor, to_mx
+
+
+torch.manual_seed(2)
+
+
+@pytest.mark.parametrize("hp_dtype", [torch.float32])
+@pytest.mark.parametrize("device", ["cuda", "cpu"])
+@pytest.mark.parametrize("sign", [1, -1])
+@pytest.mark.parametrize("use_stochastic_rounding", [False, True])
+def test_overflow_cast(hp_dtype, device, sign, use_stochastic_rounding):
```

Comment:
can we add these tests to test/prototype/mx_formats/test_custom_cast.py to keep the testing of MX numerics in one place?